.github/release-drafter.yml

```diff
@@ -17,7 +17,9 @@ exclude-labels:
   - 'skip-changelog'
 change-template: '* $TITLE (#$NUMBER)'
```

Comment:
```suggestion
change-template: '* $TITLE [#$NUMBER]($URL)'
```

This change should format the PRs into `[#123](https://github.com/GenericMappingTools/pygmt/pull/123)`, and so `sed` shouldn't be needed anymore!

Reply:
I think the `$URL` is pretty useful. The only con is that it makes lines much longer than before, so it would be a little difficult when we try to merge some changelog entries.

Reply:
We might be able to shorten things using Jinja2 substitutions in MyST (using `{{ }}`), though I haven't tested it yet.

References:
- https://myst-parser.readthedocs.io/en/v0.13.5/using/syntax-optional.html#substitutions-and-urls
- https://myst-parser.readthedocs.io/en/v0.13.5/develop/_changelog.html#new

Reply:
I think it won't work well on github release page.

Reply:
True, we do want to just copy and paste things across, ideally without changing anything.

Reply:
Not a big issue for me, but I'm wondering if we can change each entry to two lines
```
change-template: > 
    * $TITLE 
      ([#$NUMBER]($URL))
```
so it will be:
```
* Improve how PyGMT finds the GMT library 
  ([#702](https://github.com/GenericMappingTools/pygmt/pull/702))
```

Reply:
Tried it in my own fork. It doesn't work:

![image](https://user-images.githubusercontent.com/3974108/108614680-aa23e800-73ca-11eb-9ce9-5abe962dc0e4.png)

Reply:
What if you add a backslash?

Reply:
No changes at all.

Reply:
Nevermind then :sweat_smile: Should I commit this change still (i.e. let release drafter do `[#$NUMBER]($URL)` directly) or stick with the current state using `sed` (i.e. so we can review using short `#123` PR numbers and then run `sed` after)?

Reply:
If we use short PR numbers (`#123`), we need to strictly follow the following steps:

1. Review and finalize all changelog entries
2. Make a backup of the current changelog (git history already does this)
3. Run `sed` to convert the PR links 
4. Merge the changelog into master
5. Find the backup at step 2 and update the github release notes

In this case, using ReST or Markdown makes no difference to us.

If we use full PR numbers and links, 

1. Review and finalize all changelog entries
2. Merge the changelog into master
3. Copy the changelog from `changes.md` and update the github release notes

The new way (full PR numbers and links) seems easier and also doesn't rely on external tools (`sed`).

Reply:
Ok, go with the new way then (`[#$NUMBER]($URL)`). Great that @willschlitzer was clever enough to do a backup of the changelog for the PyGMT v0.3.0 release, but let's not count on our luck so much!

P.S. I'm drafting the new simplified Release Note curation/editing process in MAINTENANCE.md at https://github.com/GenericMappingTools/pygmt/pull/907/files#diff-87a0c4623bffa64c60b62a39eeab1c40bd0f329cebc177f5e4492555fb5d42ae.